### PR TITLE
Adding Mode to DaprSecretStore

### PR DESCRIPTION
**Pull request by vishwahiremat**
Adding Mode to DaprSecretStore

### CODE REVIEW

File: docs/content/reference/resource-schema/link-schema/dapr-schema/dapr-secretstore/_index.md [hunk 1]
```diff
@@ -30,6 +30,7 @@ This resource will automatically create and deploy the Dapr component spec for t
 | Key  | Required | Description | Example |
 |------|:--------:|-------------|---------|
 | kind | y | The kind of the underlying secret store resource. See [Available Dapr components](#available-dapr-components) for more information. | `secretstores.azure.keyvault`
+| mode | y |The mode used to build the pub/sub resource. Options are to build automatically via 'recipe' or build manually via 'values'. Selection determines which set of fields to additionally require. | `recipe`
```
Comment (kachawla) [inline docs/content/reference/resource-schema/link-schema/dapr-schema/dapr-secretstore/_index.md]:
```suggestion
| mode | y |The mode used to build the dapr secret store resource. Options are to build automatically via 'recipe' or build manually via 'values'. Selection determines which set of fields to additionally require. | `recipe`
```

Comment (vishwahiremat) [reply docs/content/reference/resource-schema/link-schema/dapr-schema/dapr-secretstore/_index.md]: done!

**Review by kachawla — DISMISSED** [review]

File: docs/content/reference/resource-schema/link-schema/dapr-schema/dapr-secretstore/_index.md [hunk 2]
```diff
@@ -30,6 +30,7 @@ This resource will automatically create and deploy the Dapr component spec for t
 | Key  | Required | Description | Example |
 |------|:--------:|-------------|---------|
 | kind | y | The kind of the underlying secret store resource. See [Available Dapr components](#available-dapr-components) for more information. | `secretstores.azure.keyvault`
+| mode | y |The mode used to build the secret store resource. Options are to build automatically via 'recipe' or build manually via 'values'. Selection determines which set of fields to additionally require. | `recipe`
```
Comment (AaronCrawfis) [inline docs/content/reference/resource-schema/link-schema/dapr-schema/dapr-secretstore/_index.md]:
```suggestion
| mode | y | The mode used to build the secret store resource. Options are to build automatically via 'recipe' or build manually via 'values'. Selection determines which set of fields to additionally require. | `recipe`
```

File: docs/content/reference/resource-schema/link-schema/dapr-schema/dapr-secretstore/snippets/dapr-secretstore-generic.bicep [hunk 3]
```diff
@@ -17,6 +17,7 @@ resource secretstore 'Applications.Link/daprSecretStores@2022-03-15-privateprevi
     environment: environment
     application: app.id
     kind: 'generic'
+    mode: 'values'
```
Comment (AaronCrawfis) [inline docs/content/reference/resource-schema/link-schema/dapr-schema/dapr-secretstore/snippets/dapr-secretstore-generic.bicep]:
```suggestion
    mode: 'values'
    kind: 'generic'
```

**Review by AaronCrawfis — APPROVED** [review]

**Pull request merged**